packages/components/src/internal/components/domainproperties/DateTimeFieldOptions.tsx

```diff
@@ -1,86 +1,248 @@
+
+                    if (prevSetting.isDate && !prevSetting.isTime) updates.dateFormat = parentFormat;
+                    else if (!prevSetting.isDate && prevSetting.isTime) updates.timeFormat = parentFormat;
+                    else {
+                        const parts = splitDateTimeFormat(parentFormat);
+                        updates.dateFormat = parts[0];
+                        updates.timeFormat = parts[1];
+                    }
+                }
+                updates.valid = isValidDateTimeSetting({ ...prevSetting, ...updates } as DateTimeSettingProp);
+                return {
+                    ...prevSetting,
+                    ...updates,
+                };
+            });
+
+            if (checked) onChange(domainFieldId, null);
+        },
+        [setSetting, domainFieldId, onChange]
+    );
+
+    const onFormatChange = useCallback(
+        (newFormat: string, isTime?: boolean): void => {
+            setSetting(prevSetting => {
+                const updates: Partial<DateTimeSettingProp> = {
+                    [isTime ? 'timeFormat' : 'dateFormat']: newFormat == null ? '' : newFormat,
+                };
+                updates.valid = isValidDateTimeSetting({ ...prevSetting, ...updates } as DateTimeSettingProp);
+
+                const updatedSetting = {
+                    ...prevSetting,
+                    ...updates,
+                };
+                onChange?.(domainFieldId, getDateTimeSettingFormat(updatedSetting));
+                return updatedSetting;
+            });
+        },
+        [domainFieldId, onChange]
+    );
+
+    const onDateFormatChange = useCallback(
+        (name: string, selectedValue: string, selectedOption: SelectInputOption): void => {
+            onFormatChange(selectedOption?.value);
+        },
+        [domainFieldId, onChange]
+    );
 
-    render(): ReactNode {
-        const { index, label, format, lockType, domainIndex, type } = this.props;
+    const onTimeFormatChange = useCallback(
+        (name: string, selectedValue: string, selectedOption: SelectInputOption): void => {
+            onFormatChange(selectedOption?.value, true);
+        },
+        [domainFieldId, onChange]
+    );
 
-        const noun = type === 'time' ? 'Times' : 'Dates';
-        return (
-            <div>
-                <div className="row">
-                    <div className="col-xs-12">
-                        <SectionHeading title={label} />
+    if (!setting) return null;
+
+    return (
+        <div>
+            <div className="row">
+                <div className="col-xs-12">
+                    <SectionHeading title={label} />
+                </div>
+            </div>
+            <div className="row">
+                <div className="col-xs-3">Use Default</div>
+                <div className="col-xs-9">
+                    <input
+                        checked={setting.inherited}
+                        onChange={onToggleInherited}
```

Comment:
in quick testing locally, it looks like changing the use default checkbox state isn't enough to put the page into a dirty state. maybe that is intentional

Reply:
fixed